coordinator/src/db/user.rs

```diff
@@ -36,13 +61,19 @@ impl From<RegisterParams> for User {
             fcm_token: "".to_owned(),
             last_login: OffsetDateTime::now_utc(),
             version: value.version,
+            // TODO: this is not ideal, we shouldn't need to do this as it's autogenerated in the db
+            // However, this is needed here because we convert from `RegisteredUser` to `User`. We
+            // should not do this anymore.
+            referral_code,
```

Comment:
instead of loading it from the database you could apply the same logic here, since you have the pubkey at hand.